desafio-02/rocketguedes/python/sieve.py

```diff
@@ -0,0 +1,41 @@
+"""Sieve of Eratosthenes.
+
+Find all the prime numbers less than or equal to a given integer n by Eratosthenes'
+method.
+"""
+
+import sys
+
+
+def sieve_of_eratosthenes(limit: int) -> list[int]:
+    """Return all primes smaller than or equal to N.
+
+    (including optimization of starting from prime's square)
+    """
+
+    is_prime: list[bool] = [True for _ in range(limit + 1)]
+    is_prime[0], is_prime[1] = False, False
+
+    prime: int = 2
+
+    while prime*prime <= limit:
+        if is_prime[prime]:
+            for i in range(prime*prime, limit+1, prime):
+                is_prime[i] = False
+
+        prime += 1
+
+    return [number for number, is_prime in enumerate(is_prime) if is_prime]
+
+
+if __name__ == "__main__":
+    number: int = 10000
+
+    if len(sys.argv) > 1:
+        try:
```

Comment:
Se eu passo um número negativo na linha de comando dá erro:
python3 primos2.py -35
Traceback (most recent call last):
  File "/home/marcelo/testes-desafios/primos2.py", line 41, in <module>
    print(sieve_of_eratosthenes(number))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marcelo/testes-desafios/primos2.py", line 17, in sieve_of_eratosthenes
    is_prime[0], is_prime[1] = False, False
    ~~~~~~~~^^^
IndexError: list assignment index out of range

Reply:
This issue has been fixed in commit dba039c.